### PR TITLE
chore: solc zkvm support

### PR DESCRIPTION
**Pull request by kiriyaga-txfusion**
# What :computer: 
* Allowing users to compile contracts with the zkVm solc compiler via hardaht-zksync-solc hardhat plugin

# Why :hand:
* hardaht-zksync-solc hardhat plugin have a new feature that allows to compile contracts with zkVm solc compiler and that new feature needs to be explained to users.

### CODE REVIEW

File: docs/build/tooling/hardhat/hardhat-zksync-solc.md [hunk 1]
```diff
@@ -109,6 +109,32 @@ For Solidity versions older than 0.8, only this compilation mode is available an
 
 :::
 
+For usage of zkVm compiler, `eraVersion` should be added inside `solidity` property in the `hardhat.config.ts` file:
```
Comment (hedgar2017) [inline docs/build/tooling/hardhat/hardhat-zksync-solc.md:112]: zkVm -> zkVM?

Should we now rename zkEVM and EraVM to zkVM everywhere?

Comment (kiriyaga-txfusion) [reply docs/build/tooling/hardhat/hardhat-zksync-solc.md]: Hello @hedgar2017,

I wanted to inform you that we have made a transition from zkVM to EraVM, as documented in the [official repository] (https://github.com/matter-labs/compiler-solidity). Please review the updated information to ensure that all details are accurate. Thanks!

Comment (hedgar2017) [reply docs/build/tooling/hardhat/hardhat-zksync-solc.md]: @kiriyaga-txfusion that repo is already archived and obsolete in favor of [the open-source one](https://github.com/matter-labs/era-compiler-solidity).
However, we still haven't synced across the whole docs regarding this naming, so I leave this decision to the doc owners.

File: docs/build/tooling/hardhat/hardhat-zksync-solc.md [hunk 2]
```diff
@@ -109,6 +109,34 @@ For Solidity versions older than 0.8, only this compilation mode is available an
 
 :::
 
+### zkSync Era Solidity compiler
+
+Due to the identified limitations in the [Solidity compiler](https://docs.zksync.io/zk-stack/components/compiler/toolchain/solidity.html#limitations), our team has developed a new [zkSync Era compiler](https://github.com/matter-labs/era-solidity) known as EraVM, which effectively addresses and resolves these constraints.
```
Comment (MexicanAce) [inline docs/build/tooling/hardhat/hardhat-zksync-solc.md:114]: @hedgar2017 Is `https://github.com/matter-labs/era-solidity` the correct GitHub repo to reference here?

Comment (MexicanAce) [reply docs/build/tooling/hardhat/hardhat-zksync-solc.md:114]: or should it be `https://github.com/matter-labs/era-compiler-solidity`?

Comment (hedgar2017) [reply docs/build/tooling/hardhat/hardhat-zksync-solc.md:114]: @MexicanAce it's the 1st one, but I would rewrite this paragraph from scratch:

Due to [the identified limitations](https://docs.zksync.io/zk-stack/components/compiler/toolchain/solidity.html#limitations) of the [upstream Solidity compiler](https://github.com/ethereum/solidity), our team has developed [a new edition](https://github.com/matter-labs/era-solidity) of the compiler, which effectively addresses and resolves these constraints.

Comment (kiriyaga-txfusion) [reply docs/build/tooling/hardhat/hardhat-zksync-solc.md]: Thanks @hedgar2017 for the suggestion. In the most recent commit, I have changed the paragraph accordingly with your recommendation.

**Review by MexicanAce — APPROVED** [review]
LGTM

**Pull request merged**